consensus/dbft/dbft.go

```
@@ -2175,6 +2222,24 @@ func (c *DBFT) getValidators(blockNum *uint64, state *state.StateDB, header *typ
 	return res, err
 }
 
-func (c *DBFT) shouldUpdateCommitteeAt(blockNum uint64) bool {
-	return blockNum%uint64(len(c.config.StandByValidators)) == 0
+// getDKGIndex returns validator dkg index (original validator index +1) by validatorIndex (ordered validator index).
+func (c *DBFT) getDKGIndex(validatorIndex int, blockNum uint64) (int, error) {
+	originValidators, err := c.getOriginalValidators(&blockNum, nil, nil)
+	if err != nil {
+		return -1, err
+	}
+	if validatorIndex < 0 || validatorIndex >= len(originValidators) {
+		return -1, fmt.Errorf("invalid validator index: validators count is %d, requested %d", len(originValidators), validatorIndex)
+	}
+	orderedValidators := make([]common.Address, len(originValidators))
+	copy(orderedValidators, originValidators)
+	slices.SortFunc(orderedValidators, common.Address.Cmp)
+	addr := orderedValidators[validatorIndex]
+	for i := range originValidators {
+		if orderedValidators[i] == addr {
+			return i + 1, nil
+		}
+	}
+	// impossible case
+	return -1, nil
```

Panic here instead of silent return. It's a program bug if execution flow reaches this place.